kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -418,24 +418,21 @@ FindGlobalEgIdResult KvbAppFilter::findGlobalEventGroupId(uint64_t external_even
   // At this point, we exhausted all private entries => it has to be a public event group
   uint64_t global_eg_id;
 
+  // If all private event groups were pruned then we need to adjust current_* to point to the last private event group
+  if (private_start == 0) {
+    ConcordAssertNE(private_end, 0);
+    current_pvt_eg_id = private_end;
+    current_ext_eg_id = private_end + public_start - 1;
+  }
+
   if (external_event_group_id < current_ext_eg_id) {
-    if (current_pvt_eg_id == private_start) {
-      std::tie(global_eg_id, std::ignore) = getValueFromTagTable(kPublicEgId, external_event_group_id);
-      return {global_eg_id, true, current_pvt_eg_id - 1, external_event_group_id};
-    }
     auto num_pub_egs = current_ext_eg_id - current_pvt_eg_id;
```

Comment:
Consider a scenario, wherein -
private_start = 0, private_end = 5, public_start = 1, public_end = 5, i.e., all private event groups have been pruned. Requested external_event_group_id = 1.
Now, acc. to this change, current_pvt_eg_id = 5, current_ext_eg_id = 5.
and, num_pub_egs = 0 and pub_eg_id = -3, is that expected, or am I missing something?